x-pack/plugin/security/qa/operator-privileges-tests/src/main/java/org/elasticsearch/xpack/security/operator/actions/RestGetActionsAction.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.operator.actions;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.support.TransportAction;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.SuppressForbidden;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.elasticsearch.rest.RestRequest.Method.GET;
+
+public class RestGetActionsAction extends BaseRestHandler {
+    @Override
+    public List<Route> routes() {
+        return List.of(new Route(GET, "/_test/get_actions"));
+    }
+
+    @Override
+    public String getName() {
+        return "test_get_actions";
+    }
+
+    @SuppressForbidden(reason = "Use reflection for testing only")
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        final Map<ActionType, TransportAction> actions = AccessController.doPrivileged(
+            (PrivilegedAction<Map<ActionType, TransportAction>>) () -> {
+                try {
+                    final Field actionsField = client.getClass().getDeclaredField("actions");
+                    actionsField.setAccessible(true);
+                    return (Map<ActionType, TransportAction>) actionsField.get(client);
+                } catch (NoSuchFieldException | IllegalAccessException e) {
+                    throw new ElasticsearchException(e);
+                }
```

Comment:
The earlier version is to have Guice injection for the [transport action](https://github.com/elastic/elasticsearch/blob/892840f780d89465a78a55ca4343e7aaee2a1b38/x-pack/qa/operator-privileges-tests/src/main/java/org/elasticsearch/example/actions/TransportGetActionsAction.java#L27-L31). It comes with additional bolierplate code for transportAction handling which the reflection version helps to remove. I don't mind changing back to the Guice approach again, but I probably prefer reflection because:
* It is more concise
* The team wants to get rid of Guice. Though it may take a long time, Guice is on its way out. In contrast, we do have support for Plugins to use reflection with Java policy file. So it is more formal if you ask me.

The downside is that reflection relies on `NodeClient`'s internals, which could change without notice. But since these are test code, I am not too worried.

Other than these two approach, I cannot think of anything else without touching the production code.